applications/GeoMechanicsApplication/custom_strategies/schemes/generalized_newmark_scheme.hpp

```diff
@@ -0,0 +1,88 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#pragma once
+
+#include "geomechanics_time_integration_scheme.hpp"
+
+namespace Kratos {
+
+template <class TSparseSpace, class TDenseSpace>
+class GeneralizedNewmarkScheme
+    : public GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace> {
+public:
+    explicit GeneralizedNewmarkScheme(double theta,
+                                      const Variable<double>& rVariable,
+                                      const Variable<double>& rDeltaTimeVariable,
+                                      const Variable<double>& rDeltaTimeVariableCoefficient)
+        : GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace>(),
+          mTheta(theta),
+          mVariable(rVariable),
+          mDeltaTimeVariable(rDeltaTimeVariable),
+          mDeltaTimeVariableCoefficient(rDeltaTimeVariableCoefficient)
+    {
+        KRATOS_ERROR_IF(this->mTheta <= 0)
+            << "Theta must be larger than zero, but got " << this->mTheta << "\n";
+    }
+
+protected:
+    inline void UpdateVariablesDerivatives(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        block_for_each(rModelPart.Nodes(), [&](Node& rNode) {
+            this->UpdateScalarTimeDerivative(rNode, mVariable, mDeltaTimeVariable);
+        });
```

Comment:
I have compared this implementation with the one in `BackwardEulerScheme::UpdateVariablesDerivatives` and I would make them identical:
```suggestion
        block_for_each(rModelPart.Nodes(), [this](Node& rNode) {
            this->UpdateScalarTimeDerivative(rNode, mVariable, mDeltaTimeVariable);
        });
```